chapter09/pxd/ex2.pl

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+use feature "say";
+
+
+if ( !defined $ARGV[0] ) {
+  die( "Provide file name as argument to proceed...... \n" );
+}
+
+
+$^I = '.bkp';
+
+while (<>) {
+	s/fred/Larry/gi;
+	say;
+}
+
+
+rename ("$ARGV", "$ARGV.out");
+
```

Comment:
Just realised, the original filename now has <code>.bkp</code> and should have changed it back to the original.